extract_endpoint/src/extract_endpoint/azure_utils.py

```diff
@@ -17,16 +18,66 @@ class EnvVariables(enum.Enum):
     domain = 'EXTRACT_ENDPOINT_STORAGE_DOMAIN'
 
 
-def upload_bytes_to_azure(coords: StorageCoordinates, data: bytes, filename: str) -> bool:
-    azure_service = blob.BlockBlobService(account_name=coords.account,
-                                          account_key=coords.key,
-                                          custom_domain=coords.domain)
-    azure_service.create_blob_from_bytes(coords.container, filename, data)
-    return filename in [blob.name for blob in azure_service.list_blobs(coords.container)]
+class StorageProtocol(typing_extensions.Protocol):
```

Comment:
maybe `write` and `read` would be better method names (ie more applicable to both cloud and local)

Reply:
though if we just have a MockStorage as the alternative, upload and download are better 👍

Reply:
though if we just have a MockStorage as the alternative, upload and download are better 👍